### DISCUSSION TITLE
removeOnComplete not working with age set

### DISCUSSION THREAD

**Discussion by DashBarkHuss** [taskforcesh/bullmq#1385] [Q&A]
I'm trying to delete jobs after a certain age.

I though this would delete the job 5 seconds after completed. 

```
  const job = await queue.add(
    'postWishlistItem',
    { user, body },
    {
      removeOnComplete: { age: 5 },
      removeOnFail: { age: 5 },
    }
  );
  ```
  
  But 15 minutes have passed and the job is still on my bull dashboard  in the "completed" tab showing these options:
  
  ```
  {
  "attempts": 3,
  "delay": 0,
  "backoff": {
    "delay": 1000,
    "type": "exponential"
  },
  "removeOnFail": {
    "age": 5
  },
  "removeOnComplete": {
    "age": 5
  }
}
```

Am I following the [docs](https://github.com/OptimalBits/bull/pull/2275/commits/809274089545e312a9fe8e4323836b0ace326a27) incorrectly?

**Comment by roggervalf** [comment]
hi @DashBarkHuss , I think this is related to https://github.com/taskforcesh/bullmq/issues/1353

**Reply by DashBarkHuss** [reply]
Ah yes that's it. It seems to always leave one in there but will delete it when the next one comes. Not sure the purpose of that but it works for me!

**Reply by manast** [reply]
It is because the removal is lazy, so when a new job is completed/failed it will remove jobs that are older than the given setting.
